Converting to a draft for the moment... I somehow snuck through successful local testing in between `npm` runs, but starting from scratch shows that the `custom-card-helpers` upgrade propagates more deeply... This change would (i think?) require updating `custom-card-helpers` in `lovelace-timer-bar-card` as well, in order to propagate our `HomeAssistant` instance from one card to the next...

It's possible that my newbie level JavaScript (and TypeScript) is simply missing some trick here, but at the moment I get errors like: 
```
Error: /workspaces/opensprinkler-card/src/opensprinkler-control.ts(62,23): semantic error TS2345: Argument of type 'import("/workspaces/opensprinkler-card/node_modules/custom-card-helpers/dist/types").HomeAssistant' is not assignable to parameter of type 'import("/workspaces/opensprinkler-card/node_modules/lovelace-timer-bar-card/node_modules/custom-card-helpers/dist/types").HomeAssistant'.
  Types of property 'auth' are incompatible.
    Type 'import("/workspaces/opensprinkler-card/node_modules/custom-card-helpers/node_modules/home-assistant-js-websocket/dist/auth").Auth' is not assignable to type 'import("/workspaces/opensprinkler-card/node_modules/lovelace-timer-bar-card/node_modules/home-assistant-js-websocket/dist/auth").Auth'.
      Types have separate declarations of a private property '_saveTokens'.
```

@rianadon Is it preferable here to propagate this upgrade onward into the timer bar card as well? Or, should I drop the `custom-card-helpers` dependency back down to 1.8.0 and sort out the relative time some other way? 

The hang up on 1.8.0 is that something is going sideways in `localize` and I end up with empty strings out the far end of `relativeTime`. Others have reported that issue, but it's fixed with the move to locale data in 1.9.0.

--- a/README.md
+++ b/README.md
@@ -38,6 +38,8 @@ Otherwise, make sure:
 - The ids of program running binary sensors end with `_running`
 - The id of the OpenSprinkler controller enabled switch ends with `_enabled`
 - The ids of program & station enabled switches end with `_enabled`
+- The id of the rain delay active binary sensor ends with `_rain_delay_active`
+- The id of the rain delay stop time sensor ends with `_rain_delay_stop_time`
 
 ## Extra entities and duration control
 

--- a/package.json
+++ b/package.json
@@ -14,9 +14,10 @@
   "repository": "git@github.com:rianadon/opensprinkler-card.git",
   "license": "Apache-2.0",
   "dependencies": {
+    "@formatjs/intl-utils": "^3.8.4",
     "@mdi/js": "^6.5.95",
-    "custom-card-helpers": "^1.9.0",
-    "home-assistant-js-websocket": "^6.0.1",
+    "custom-card-helpers": "1.8.0",
+    "home-assistant-js-websocket": "^5.12.0",
     "lit": "^2.0.2",
     "lovelace-timer-bar-card": "^1.15.0"
   },

--- a/src/opensprinkler-card.ts
+++ b/src/opensprinkler-card.ts
@@ -1,6 +1,6 @@
 import { LitElement, css, html, TemplateResult } from 'lit';
 import { customElement, state, property } from "lit/decorators";
-import { HomeAssistant, LovelaceCardEditor, relativeTime } from 'custom-card-helpers';
+import { HomeAssistant, LovelaceCardEditor } from 'custom-card-helpers';
 import { PropertyValues } from 'lit-element';
 import { UnsubscribeFunc } from 'home-assistant-js-websocket';
 
@@ -16,6 +16,7 @@ import { MoreInfoDialog } from './opensprinkler-more-info-dialog';
 import { EntitiesFunc, hasManual, hasRainDelayActive, hasRunOnce, isPlayPausable, isProgram, isRainDelayStopTime, isStation, lineHeight, osName, stateActivated, stateWaiting } from './helpers';
 import { renderState } from './opensprinkler-state';
 import { styleMap } from 'lit/directives/style-map';
+import { relativeTime } from './relative_time';
 
 // This puts your card into the UI card picker dialog
 (window as any).customCards = (window as any).customCards || [];
@@ -211,7 +212,7 @@ export class OpensprinklerCard extends LitElement {
 
     if (hasRainDelayActive(entities)) {
       const stop_time = entities(isRainDelayStopTime).find(_ => true)?.state;
-      return `Rain delay${ stop_time ? ` ends ${relativeTime(new Date(stop_time), this.hass!.locale)}` : ''}`;
+      return `Rain delay${ stop_time ? ` ends ${relativeTime(new Date(stop_time), this.hass!.locale!)}` : ''}`;
     }
 
     const programs = entities(isProgram).filter(stateActivated).map(osName);

--- a/src/relative_time.ts
+++ b/src/relative_time.ts
@@ -1,0 +1,52 @@
+//REF: https://github.com/custom-cards/custom-card-helpers/blob/master/src/datetime/relative_time.ts
+//REF: https://github.com/home-assistant/frontend/blob/dev/src/common/datetime/relative_time.ts
+
+import { selectUnit } from "@formatjs/intl-utils";
+
+// REF: https://github.com/custom-cards/custom-card-helpers/blob/master/src/types.ts
+// REF: https://github.com/home-assistant/frontend/blob/dev/src/data/translation.ts
+enum NumberFormat {
+    language = "language",
+    system = "system",
+    comma_decimal = "comma_decimal",
+    decimal_comma = "decimal_comma",
+    space_comma = "space_comma",
+    none = "none",
+}
+
+export enum TimeFormat {
+    language = "language",
+    system = "system",
+    am_pm = "12",
+    twenty_four = "24",
+}
+
+interface MinBarFrontendLocaleData {
+    language: string;
+}
+
+ const formatRelTimeMem =
+  (locale: MinBarFrontendLocaleData) =>
+    new Intl.RelativeTimeFormat(locale.language, { numeric: "auto" });
+
+/**
+ * Calculate a string representing a date object as relative time from now.
+ *
+ * Example output: 5 minutes ago, in 3 days.
+ */
+ export const relativeTime = (
+  from: Date,
+  locale: MinBarFrontendLocaleData,
+  to?: Date,
+  includeTense = true
+): string => {
+  const diff = selectUnit(from, to);
+  if (includeTense) {
+    return formatRelTimeMem(locale).format(diff.value, diff.unit);
+  }
+  return Intl.NumberFormat(locale.language, {
+    style: "unit",
+    unit: diff.unit,
+    unitDisplay: "long",
+  }).format(Math.abs(diff.value));
+};

